### PR TITLE
TFMA notebook showing ExtractEvaluateAndWriteResult Tranfsorm

### PR DESCRIPTION
**Pull request by Juta**
TFMA notebook
Next PR: website content about ML evaluation + link to this notebook

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Juta** [comment]
@damccorm

File: examples/notebooks/beam-ml/tfma_beam.ipynb [hunk 1]
```diff
@@ -0,0 +1,669 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# TensorFlow Model Analysis in Beam\n",
+        "[TensorFlow Model Analysis (TFMA)](https://www.tensorflow.org/tfx/guide/tfma) is a library for performing model evaluation across different slices of data. TFMA performs its computations in a distributed manner over large amounts of data using Apache Beam.\n",
+        "\n",
+        "This example colab notebook illustrates how TFMA can be used to investigate and visualize the performance of a model as part of your Beam pipeline. This allows for scalable and flexible execution of your evaluation pipeline. For this we will use [**ExtractEvaluateAndWriteResults**](https://www.tensorflow.org/tfx/model_analysis/api_docs/python/tfma/ExtractEvaluateAndWriteResults), which is a PTransform for performing extraction, evaluation, and writing results all in one step.\n",
```
Comment (damccorm) [inline examples/notebooks/beam-ml/tfma_beam.ipynb:9]:
```suggestion
        "This example notebook illustrates how TFMA can be used to investigate and visualize the performance of a model as part of your Beam pipeline. This allows for scalable and flexible execution of your evaluation pipeline. For this we will use [**ExtractEvaluateAndWriteResults**](https://www.tensorflow.org/tfx/model_analysis/api_docs/python/tfma/ExtractEvaluateAndWriteResults), which is a PTransform for performing extraction, evaluation, and writing results all in one step.\n",
```

Nit: Doesn't have to be colab, right?

File: examples/notebooks/beam-ml/tfma_beam.ipynb [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Visualize results\n",
+        "results = tfma.load_eval_results(output_paths=output_path_compare)\n",
+        "tfma.view.render_time_series(results)"
```
Comment (damccorm) [inline examples/notebooks/beam-ml/tfma_beam.ipynb:644]: Right now, this doesn't render in GitHub - https://github.com/apache/beam/blob/50e2a440729ea19ae939584933d96e11b156bf6d/examples/notebooks/beam-ml/tfma_beam.ipynb (as compared to the results in https://github.com/apache/beam/blob/50e2a440729ea19ae939584933d96e11b156bf6d/examples/notebooks/beam-ml/run_inference_multi_model.ipynb for example)

I think if you run the notebook and upload it with the run results it should render - could you do that?

**Review by damccorm — COMMENTED** [review]
This mostly looks good to me outside of a couple small comments. @rszper could you please take a look?

File: examples/notebooks/beam-ml/tfma_beam.ipynb [hunk 3]
```diff
@@ -0,0 +1,669 @@
+{
```
[the remark this answers is not in the document]
Comment (damccorm) [reply examples/notebooks/beam-ml/tfma_beam.ipynb:1]: Yes, good catch. That is causing the precommit failure

File: examples/notebooks/beam-ml/tfma_beam.ipynb (hunk 2 above)
Comment (Juta) [reply examples/notebooks/beam-ml/tfma_beam.ipynb:644]: I first committed with the output of the two visualisations. But they did not render on Github so I removed them. I can try again if you want but I don't know if/ how can can solve this

**Comment by Juta** [comment]
@damccorm @rszper Thanks for the review. I included your suggestions.

File: examples/notebooks/beam-ml/tfma_beam.ipynb (hunk 2 above)
Comment (Juta) [reply examples/notebooks/beam-ml/tfma_beam.ipynb:699]: I added screenshots for now. Suggestions on how to do it otherwise are still welcome of course

Comment (damccorm) [reply examples/notebooks/beam-ml/tfma_beam.ipynb:699]: LGTM - it looks like an extra block got added to the bottom of the notebook though, could you please remove it? Unfortunately the diff no longer renders, so a screenshot is the best I can do:

image

**Review by damccorm — APPROVED** [review]
Had one last comment, after that is addressed I will merge

File: examples/notebooks/beam-ml/tfma_beam.ipynb (hunk 2 above)
Comment (Juta) [reply examples/notebooks/beam-ml/tfma_beam.ipynb:699]: Ok thanks, I removed it

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**